### PR TITLE
[mod_tags_similar] Link cleanup

### PR DESCRIPTION
**Pull request by SharkyKZ**
Pull Request for Issue # .

### Summary of Changes
This moves link processing to helper and removes unused `JHelperRoute` instance.

### Testing Instructions
No changes in behavior, module works like before.

### Documentation Changes Required
No.

### CODE REVIEW

File: modules/mod_tags_similar/helper.php [hunk 1]
```diff
@@ -61,19 +63,19 @@ public static function getList(&$params)
...
-				$db->quoteName('cc.core_alias'),
-				$db->quoteName('cc.core_catid'),
-				$db->quoteName('cc.core_language'),
-				$db->quoteName('cc.core_params')
+					$db->quoteName('ct.router'),
+					$db->quoteName('cc.core_title'),
+					$db->quoteName('cc.core_alias'),
+					$db->quoteName('cc.core_catid'),
+					$db->quoteName('cc.core_language'),
+					$db->quoteName('cc.core_params'),
```
Comment (brianteeman) [inline modules/mod_tags_similar/helper.php:76]: I thought the last item in the array did not have a comma

Comment (Quy) [reply modules/mod_tags_similar/helper.php:76]: Per coding standards: 

> When splitting array definitions onto several lines, the last value should also have a trailing comma.

Comment (brianteeman) [reply modules/mod_tags_similar/helper.php:76]: Ah ok - confused by line 159+ then

File: modules/mod_tags_similar/tmpl/default.php [hunk 2]
```diff
@@ -21,8 +20,7 @@
 					echo htmlspecialchars($item->core_title, ENT_COMPAT, 'UTF-8');
 				endif; ?>
 			<?php else : ?>
-				<?php $item->route = new JHelperRoute; ?>
-				<a href="<?php echo JRoute::_(TagsHelperRoute::getItemRoute($item->content_item_id, $item->core_alias, $item->core_catid, $item->core_language, $item->type_alias, $item->router)); ?>">
+				<a href="<?php echo JRoute::_($item->link); ?>">
 					<?php if (!empty($item->core_title)) :
```
Comment (Quy) [inline modules/mod_tags_similar/tmpl/default.php]: Please consider changing as follows for [mixed language usage](https://developer.joomla.org/coding-standards/php-code.html#user-content-mixed-language-usage-eg-at-the-layout-files) and above lines 19-21.

```
<?php if (!empty($item->core_title)) : ?>
	<?php echo htmlspecialchars($item->core_title, ENT_COMPAT, 'UTF-8'); ?>
<?php endif; ?>
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 050952ba7bd91449a4c46d8853c03b142884f245<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20730">issues.joomla.org/tracker/joomla-cms/20730</a>.</sub>

**Comment by carlitorweb** [comment]
I have tested this item :white_check_mark: successfully on 050952ba7bd91449a4c46d8853c03b142884f245<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20730">issues.joomla.org/tracker/joomla-cms/20730</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20730">issues.joomla.org/tracker/joomla-cms/20730</a>.</sub>

File: modules/mod_tags_similar/helper.php (hunk 1 above)
Comment (mbabker) [reply modules/mod_tags_similar/helper.php:76]: That's not an array, it's a set of method parameters.  PHP doesn't allow trailing commas for those (well, maybe in PHP 7.2 but that't out of scope for a long time for us if that's true).

**Pull request merged**